### ISSUE TITLE
Rubinius: undefined method `caller_locations' on RSpec::CallerFilter

### ISSUE THREAD

**Issue by dblock** [rspec/rspec-core#1863]
Not sure if this is in rspec-core or rspec-expectations, but on Rubinius 2.2.10 on Linux I am getting these:

```
     Failure/Error: expect(File).to receive(:file?).with(path).and_return(true)
     NoMethodError:
       undefined method `caller_locations' on RSpec::CallerFilter (Class)
     # kernel/delta/kernel.rb:78:in `caller_locations (method_missing)'
     # kernel/common/kernel.rb:460:in `loop'
     # kernel/common/hash.rb:356:in `fetch'
     # ./spec/hashie/mash_spec.rb:605:in `__script__'
     # kernel/common/eval.rb:101:in `instance_exec'
     # kernel/bootstrap/array.rb:76:in `each'
     # kernel/common/array.rb:1179:in `reverse_each'
     # kernel/bootstrap/array.rb:97:in `map'
     # kernel/bootstrap/array.rb:97:in `map'
     # kernel/bootstrap/array.rb:97:in `map'
     # kernel/bootstrap/array.rb:97:in `map'
     # kernel/common/kernel.rb:447:in `load'
     # kernel/common/block_environment.rb:53:in `call_on_instance'
     # kernel/common/eval.rb:176:in `eval'
     # kernel/delta/code_loader.rb:66:in `load_script'
     # kernel/delta/code_loader.rb:152:in `load_script'
     # kernel/loader.rb:649:in `script'
     # kernel/loader.rb:825:in `main'
```

This was not the case with rspec-core 3.1.7.

https://travis-ci.org/intridea/hashie/jobs/49369452

**Comment by myronmarston** [comment]
Thanks for reporting this.  The error was happening [here](https://github.com/rspec/rspec-support/blob/v3.2.0/lib/rspec/support/caller_filter.rb#L63), in rspec-support. `caller_locations` is a new feature of Ruby 2.0.0+ that is faster than `caller` (see rspec/rspec-support#140).  Apparently rbx 2.2 claims to be `RUBY_VERSION >= '2.0.0'` but doesn't implement this API :(.  I've got a fix in rspec/rspec-support#169 that I plan to release in rspec-support 3.2.1 later today.

BTW, our rbx builds on travis don't run into this error.  We just use `rbx` in our travis build matrix (rather than `rbx-2.2.10` like you have). The current rbx version available on travis is 2.5.2 and that's what our builds run against, and it looks like `caller_locations` is in rbx 2.5.2.  Any reason you are pinning your rbx builds to an old version?

**Comment by dblock** [comment]
Because rbx-2 is a moving target, the actual Rubinius version changes as new builds are made. It's unlike MRI which makes new builds that are patches on top of the previous one, Rubinius 2.x often has big breaking changes and I got tired of chasing them :)

**Comment by myronmarston** [comment]
> Because rbx-2 is a moving target, the actual Rubinius version changes as new builds are made. It's unlike MRI which makes new builds that are patches on top of the previous one, Rubinius 2.x often has big breaking changes and I got tired of chasing them :)

Makes sense.  I can definitely relate: we've put a ton of effort into getting passing rbx builds for RSpec but it became such a PITA (with such a low ROI, given how few users use rbx) that we gave up and just have rbx in the `allowed_failures` section now.  It's pretty ridiculous how much effort it takes to keep passing builds against rbx compared to MRI or JRuby :cry:

**Comment by dblock** [comment]
Sometimes I feel like I care just because it exists, not because anyone uses it :)

For popular libraries like RSpec or Hashie I think it's important to keep at it, otherwise there's no hope for competing Ruby runtime implementations and it's good to have alternatives.

**Comment by myronmarston** [comment]
> For popular libraries like RSpec or Hashie I think it's important to keep at it, otherwise there's no hope for competing Ruby runtime implementations and it's good to have alternatives.

I think it's great that you've kept up passing builds.  We tried and simply can't -- for one, RBX frequently fails to compile simple ruby files [like this](https://github.com/rspec/rspec-expectations/blob/b5afcfbcfb0ca3ddefee1d469aa246ac7125a620/lib/rspec/matchers/built_in/match.rb) in our builds (see https://github.com/rubinius/rubinius/issues/2934).  It's been that way for over a year.  Not much we can do about that :(.

We keep passing builds against 3 different ruby implementations (JRuby, MRI 1.8.7, MRI 1.9+ -- remember that MRI 1.9 was a complete rewrite of the interpreter!) across many different versions (including 1.8 mode, 1.9 mode and 2.0 mode for JRuby) and to the extent that an alternative implementation mirrors the behavior of one of the plethora of versions we test against, it should run RSpec just fine.

**Comment by myronmarston** [comment]
I just released rspec-support 3.2.1 with a fix.
